src/lavinmq/name_validator.cr

```diff
@@ -0,0 +1,14 @@
+require "./error"
+
+class NameValidator
+  PREFIX_LIST = ["mqtt.", "amq."]
+
+  def self.valid_prefix?(name)
+    return true if PREFIX_LIST.any? { |prefix| name.starts_with? prefix }
```

Comment:
I think the return type is `Bool | Nil` now when you removed the `return false`. Simplify it to

```suggestion
    PREFIX_LIST.any? { |prefix| name.starts_with? prefix }
```